fighters/duckhunt/src/opff.rs

```diff
@@ -4,44 +4,130 @@ use super::*;
 use globals::*;
 
  
-unsafe fn duck_jump_cancel(boma: &mut BattleObjectModuleAccessor, status_kind: i32, cat1: i32, frame: f32) {
-    if status_kind == *FIGHTER_DUCKHUNT_STATUS_KIND_SPECIAL_HI_FLY {
-        if frame > 20.0 {
-            if boma.is_cat_flag(Cat1::SpecialHi) {
-                StatusModule::change_status_request_from_script(boma, *FIGHTER_DUCKHUNT_STATUS_KIND_SPECIAL_HI_END, true);
-            }
-        }
+unsafe fn duck_jump_cancel(fighter: &mut L2CFighterCommon) {
```

Comment:
this isnt a jump cancel, so it would be nice to rename it to just up special cancel or something. Also, it looks like you can only cancel this with up special (meaning you cant just cancel it with *any* special input). Is this intended?

Reply:
Calc put that there, I have not touched it. And yes, basically you can cancel into a move, or you can cancel with an up special input in order to go into freefall immediately (intentionally) and grab ledge. This is already HDR behavior. Without it, his recovery would be dogwater as there is no ledgesnapping in this game.